File: api-server/models/Appointment.js
const mongoose = require('mongoose');

const appointmentSchema = new mongoose.Schema({
    patientId: { type: String, required: true },
    patientName: { type: String, required: true },
    vetId: { type: String, required: true },
    vetName: { type: String, required: true },
    appointmentDate: { type: Date, required: true },
    appointmentType: { type: String, required: true },
    status: { type: String, default: 'Scheduled' },
    reason: { type: String },
    notes: { type: String },
    createdAt: { type: Date, default: Date.now },
    updatedAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Appointment', appointmentSchema);

File: web-frontend/Models/Appointment.cs
using System;

namespace PawsCare.Web.Models
{
    public class Appointment
    {
        public string Id { get; set; }
        public string PatientId { get; set; }
        public string PatientName { get; set; }
        public string VetId { get; set; }
        public string VetName { get; set; }
        public DateTime AppointmentDate { get; set; }
        public string AppointmentType { get; set; }
        public string Status { get; set; }
        public string Reason { get; set; }
        public string Notes { get; set; }
    }
}
